airdrop/application/services/user_registration_services.py

```diff
@@ -75,14 +75,22 @@ def eligibility(self, inputs):
                     "Airdrop window is not accepting registration at this moment"
                 )
 
-            is_eligible_user = self.check_user_eligibility()
+            is_eligible_user = self.check_user_eligibility(
+                user_address=address, airdrop_id=airdrop_id, airdrop_window_id=airdrop_window_id)
 
-            if not is_eligible_user:
-                raise Exception(
-                    "Address is not eligible for this airdrop"
-                )
+            is_already_registered = self.is_elgible_registered_user(
+                airdrop_window_id, address)
+
+            is_airdrop_window_claimed = False
+            airdrop_claim_status = self.is_airdrop_window_claimed(
+                airdrop_window_id, address)
+
+            if airdrop_claim_status is not None:
```

Comment:
`is_airdrop_window_claimed` has to be true only if the airdrop_claim_status be `SUCCESS`. 
If it is `PENDING` or `FAILED` or has not been initiated yet , the is_claimed should be false.